diff -ru cexcept-2-0-1/cexcept.h cexcept-2-0-2-optipng/cexcept.h
--- cexcept-2-0-1/cexcept.h	2008-07-23 22:35:51 +0300
+++ cexcept-2-0-2-optipng/cexcept.h	2022-11-20 12:53:12 +0200
@@ -1,13 +1,16 @@
 /*===
-cexcept.h 2.0.1 (2008-Jul-19-Sat)
+cexcept.h 2.0.2-optipng (2022-Nov-20)
+Updated by Cosmin Truta
+
 http://www.nicemice.net/cexcept/
+
 Adam M. Costello
 http://www.nicemice.net/amc/
 
 An interface for exception-handling in ANSI C (C89 and subsequent ISO
 standards), developed jointly with Cosmin Truta.
 
-    Copyright (c) 2000-2008 Adam M. Costello and Cosmin Truta.
+    Copyright (c) 2000-2022 Adam M. Costello and Cosmin Truta.
     This software may be modified only if its author and version
     information is updated accurately, and may be redistributed
     only if accompanied by this unaltered notice.  Subject to those
@@ -210,8 +213,9 @@
 
 #define Try \
   { \
-    jmp_buf *exception__prev, exception__env; \
-    exception__prev = the_exception_context->penv; \
+    jmp_buf *exception__prev[1]; \
+    jmp_buf exception__env; \
+    exception__prev[0] = the_exception_context->penv; \
     the_exception_context->penv = &exception__env; \
     if (setjmp(exception__env) == 0) { \
       do
@@ -223,7 +227,7 @@
     else { \
       the_exception_context->caught = 1; \
     } \
-    the_exception_context->penv = exception__prev; \
+    the_exception_context->penv = exception__prev[0]; \
   } \
   if (!the_exception_context->caught || action) { } \
   else
diff -ru cexcept-2-0-1/README cexcept-2-0-2-optipng/README
--- cexcept-2-0-1/README	2008-07-23 22:35:51 +0300
+++ cexcept-2-0-2-optipng/README	2022-11-20 12:53:12 +0200
@@ -1,5 +1,8 @@
-cexcept: README 2.0.1 (2008-Jul-23-Wed)
+cexcept: README 2.0.2-optipng (2022-Nov-20)
+Updated by Cosmin Truta
+
 http://www.nicemice.net/cexcept/
+
 Adam M. Costello
 http://www.nicemice.net/amc/
 
@@ -14,15 +17,15 @@
 
     name        version
     ----------  -------
-    README       2.0.1
-    cexcept.h    2.0.1
+    README       2.0.2-optipng
+    cexcept.h    2.0.2-optipng
     example1.c   2.0.0
     example2.c   2.0.0
     rationale    2.0.0
 
 The package is both free-as-in-speech and free-as-in-beer:
 
-    Copyright (c) 2000-2008 Adam M. Costello and Cosmin Truta.
+    Copyright (c) 2000-2022 Adam M. Costello and Cosmin Truta.
     This package may be modified only if its author and version
     information is updated accurately, and may be redistributed
     only if accompanied by this unaltered notice.  Subject to those
@@ -49,6 +52,21 @@
 The remainder of this file is a log of changes.
 
 
+2022-Nov-20-Sun
+
+  cexcept.h 2.0.1-optipng --> 2.0.2-optipng
+    * Improved the workaround against aggressive optimizing compilers
+      that might clobber the exception__prev pointer. Instead of making
+      this pointer volatile, we are boxing it inside a one-element array.
+
+
+2011-Jul-22-Fri
+
+  cexcept.h 2.0.1 --> 2.0.1-optipng
+    * Made exception__prev volatile to avoid "variable might be clobbered
+      by longjmp" warnings when a function contains multiple Try blocks.
+
+
 2008-Jul-23-Wed
 
   cexcept.h 2.0.0 --> 2.0.1
@@ -62,6 +80,7 @@
     * Reworded the license to agree with cexcept.h.
     * Updated the URLs.
 
+
 2001-Jul-12-Thu
 
   cexcept.h 1.0.0 --> 2.0.0
