diff -ru zlib-1.3/zlib.h zlib-1.3-optipng/zlib.h
--- zlib-1.3/zlib.h	2023-08-18 11:45:36 +0300
+++ zlib-1.3-optipng/zlib.h	2023-08-19 13:15:28 +0300
@@ -1,3 +1,25 @@
+/*
+ * zlib version 1.3-optipng, 2023-Aug-19
+ *
+ * This is a modified version of zlib.
+ * See README.optipng.txt for more information.
+ */
+
+#ifndef OPTIPNG_CONFIG_ZLIB
+#define OPTIPNG_CONFIG_ZLIB
+
+#ifndef ZLIB_CONST
+#define ZLIB_CONST
+#endif
+
+#define NO_GZCOMPRESS
+#define NO_GZIP
+
+#define TOO_FAR 32768U
+
+#endif /* OPTIPNG_CONFIG_ZLIB */
+
+
 /* zlib.h -- interface of the 'zlib' general purpose compression library
   version 1.3, August 18th, 2023
 
@@ -37,12 +59,12 @@
 extern "C" {
 #endif
 
-#define ZLIB_VERSION "1.3"
-#define ZLIB_VERNUM 0x1300
+#define ZLIB_VERSION "1.3-optipng"
+#define ZLIB_VERNUM 0x130f
 #define ZLIB_VER_MAJOR 1
 #define ZLIB_VER_MINOR 3
 #define ZLIB_VER_REVISION 0
-#define ZLIB_VER_SUBREVISION 0
+#define ZLIB_VER_SUBREVISION 15
 
 /*
     The 'zlib' compression library provides in-memory compression and
